### DISCUSSION TITLE
Enhancement: Plugins - Your licence has expired!

### DISCUSSION THREAD

**Discussion by richhayler** [craftcms/cms#8664] [Ideas]
It would be better if the individual license expired notifications for plugins were less obtrusive in the updates section of the control panel. Alternatively a way to acknowledge the expiry and dismiss it or set to be reminded in a month or when new updates are available for the expired plugin. 

It’s annoying that these messages for plugins are displayed in amongst licensed plugin updates. Maybe a one liner at the top to mention you have plugins with expired licenses and a link to the plugins section to see which ones (I don’t think we want the persistent alert notification bar that displays on all CP pages)

For a few plugins that provide basic functionality there may be no pressing need to extend the license immediately. So it’s annoying to be reminded every time I want to view updates for other plugins I have to see these messages. Furthermore it then seems unnecessary to list the individual expired plugin notifications in amongst other plugin updates. 

Please can the UI and notification or indication a license has expired for a plugin be reconsidered on the updates section.

**Comment by brandonkelly** [comment]
FWIW, those messages only show up if an update is available.

**Comment by richhayler** [comment]
That doesn’t appear to be the case unless I’m misunderstanding something. 

![589B9650-B93F-4000-BBBC-3864B22A89CC](https://user-images.githubusercontent.com/2505643/56960949-c1f70000-6b95-11e9-8572-54c721cdbcbb.jpeg)
![4752A39B-492C-4DC6-8FA4-58BA809BF54C](https://user-images.githubusercontent.com/2505643/56960950-c1f70000-6b95-11e9-84d1-fb90cea52eea.jpeg)
![2E41CC30-6B31-4C7D-8690-0B2AC8A64BC4](https://user-images.githubusercontent.com/2505643/56960951-c28f9680-6b95-11e9-8dfd-1ebcf665a263.jpeg)
![AB00B477-B46E-4916-92CC-23B6F89EDA01](https://user-images.githubusercontent.com/2505643/56961055-0aaeb900-6b96-11e9-9298-723353eafe34.jpeg)
![DA0AFFFC-3FCE-4A49-A5C8-031B72C1EB37](https://user-images.githubusercontent.com/2505643/56961057-0aaeb900-6b96-11e9-899e-b31905848efb.jpeg)

**Comment by richhayler** [comment]
Not to be pedantic but I also think the alert messaging could be a little misleading “another year of amazing updates”, as what commitment does the developer make that updates will continue to come through despite paying the renewal fee. When purchasing a plugin it’s clear what you are paying for. Maybe on the updates section it would be better to have a separate tab for expired plugins and list all the updates for each since the license expired so an admin can “scan” for relevant updates that entice them to renew but this info doesn’t get in the way of licensed plugin updates.

**Comment by brandonkelly** [comment]
> That doesn’t appear to be the case unless I’m misunderstanding something.

Sorry, you’re right. It was the case for the `./craft update` command, but not the Updates utility apparently. Just fixed that for the next release so only plugins with available updates will get listed.

> Not to be pedantic but I also think the alert messaging could be a little misleading “another year of amazing updates”, as what commitment does the developer make that updates will continue to come through despite paying the renewal fee.

After a year of using a plugin I imagine you should have a good idea of what sort of update frequency you can expect from the plugin, etc. And you can always scan its changelog to jog your memory.

> Maybe on the updates section it would be better to have a separate tab for expired plugins and list all the updates for each since the license expired so an admin can “scan” for relevant updates that entice them to renew but this info doesn’t get in the way of licensed plugin updates.

That’s the general idea here, but not sure how I feel about moving that info to a separate tab where it will likely be overlooked.

**Comment by richhayler** [comment]
Thanks for all your responses and comments above. I think the change to only show a plugin with an expired license when there are updates for that plugin is a good step forward. It will be interesting to see if anyone else has commentary / suggestions about this going forward. The annual plugin license renewals are generally all very reasonable from a cost perspective so I’m sure I’ll be renewing most and setting this as the default option but I am considering the scenario where some plugins are intentionally not renewed and the alerts are acknowledged but there is no way to dismiss them or remove them from the list of licensed plugin updates.

**Comment by richhayler** [comment]
On further reflection I do believe we need an option to optionally “mute” an expired plugin. 

What “mute” means for a CraftCMS admin:

“I’m acknowledging the plugin license subscription has expired and I’m intentionally choosing to not renew it at this time. I understand that this means I can continue to use the plugin features but I won’t be eligible for any future updates.”
By choosing to “mute” this plugin I expect that:
- Updates for the expired plugin will not show up in the same “utilities > updates” feed for plugins with active subscriptions. 
- The “updates” dashboard widget will exclude muted plugins from the “updates available” status. 

If there is no intention to renew the subscription for a plugin then it could have a detrimental effect if there is no option to mute it and the updates widget status reports there are “updates available”. Technically these could be updates that can’t be applied so there is a risk that the update status could become ignored as it’s assumed it’s for plugins that have expired and can’t be updated.

**Comment by narration-sd** [comment]
Two thoughts:

- Rather than a 'mute', what about a separate list admin can switch a  plugin to, so that there are two: 'paid updates', 'not presently updating' . 
  = When on the 'not presently' list, no flash-bangs as seems the present setup has (haven't yet tried it). Just a sensible listing of status, and advantages in upgrade. Probably the upgrade price would be sensible to have also.
  = honestly, my first impression on seeing the present listing is that it's too enthusiastic -- too much in your face. Although I well understand the sales posture after re-reading the long discussion for the store this morning. Tamping that down, no hard sell, may well actually improve buyer response?

- Second point is that with a kind of major plugin about to ship, I can't seem to find anything about security updates. My arrangement isn't prone to such problems, but we should all be thinking for them, as who knows the ways the disruptive may find to come in. My takeaway possible at the moment is that we could ship minor semver updates for free use no matter how old a purchase was. Is this true? Would the normal 'do a Craft update' or 'do a plugin update' take care of installing such? Should there be a clearly stated posture??

Thank you for your time....

**Comment by narration-sd** [comment]
What's on that commit  just showed up sounds a quite good move -- from several angles...

**Comment by proimage** [comment]
Tangentially-related, @brandonkelly, I'd recommend changing the content color in that alert from dark:

![image](https://user-images.githubusercontent.com/1027523/82047185-0418b400-96bb-11ea-9052-42da44d53e6f.png)

...to light:
![image](https://user-images.githubusercontent.com/1027523/82047190-0844d180-96bb-11ea-9397-ad7e1d6f1d0c.png)

**Comment by bencresty** [comment]
I agree with @richhayler on muting this expire-messages or at least being able to hit some button with a cross to clear the message for a particular plugin from than on.

Right now the message for an expired license stays forever and there's no way to get rid of it AFAIK. That's not very user friendly. I understand Craft wants to promote paying over and over for a plugin, but if people don't want to do that they don't want to do that for whatever reason and it's pretty annoying to have this attention-seekers in the utility panel and not giving us a chance to get rid of them. Than these attention seekers, like the attention number circle next to the Updates-menu to show there's an update, loose their value as we can't take these warnings seriously anymore.

The same goes for updates of plugins where the license has expired. There's no way to update them (and we obviously chose to not wanting to do so (yet)), but the update-messages stays in the Updates list. We should be able to remove these updates we don't want to install as we decided not to upgrade the license. So IMO there should be some button with a cross-icon to get rid of them.

When a license is expired it's obviously not the plugin that has expired. We only choose to not (yet) update it for whatever reason. I'm pretty sure people don't let them force to upgrade a plugin just because Craft keeps on shouting the plugin has expired, but they feel frustrated to not being able to get rid of these messages.

Please make it possible to mute a license-expired message, for example with a close-button. Once we clicked it away we KNOW it's expired so we don't need to see the message nor newer updates again. Also the number showing after the 'Updates' menu should be gone than as now it keeps on seeking our attention, while nothing had changed and we obviously don't want the license upgrade or we should have done it already or will do it later.
Maybe there could be a seperated list on the Updates page with only updates to expired plugins listed and leave the license message. The two emails we get before the license expired should be plenty enough.

![image](https://user-images.githubusercontent.com/69322733/124391442-ee1be780-dcf0-11eb-8d26-aca3403fbcd0.png)
